src/drt/src/db/tech/frLayer.h

```diff
@@ -675,7 +686,7 @@ class frLayer
   std::map<std::string, int> name2CutClassIdxMap;
   frCollection<std::weak_ptr<frConstraint>> constraints;
 
-  frCollection<std::weak_ptr<frLef58SpacingEndOfLineConstraint>>
+  frCollection<frLef58SpacingEndOfLineConstraint*>
```

Comment:
With this change you can also change the shared_ptr to unique_ptr in dbTechObject

Reply:
do you mean for lef58spacingeol only or all rules?

Reply:
Never mind, I see all constraints are in one container.